src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -1080,16 +1081,21 @@ private void writeLogTo(WriteMethod method, OutputStream os) throws IOException
 
     @Override public File getLogFile() {
         LOGGER.log(Level.WARNING, "Avoid calling getLogFile on " + this, new UnsupportedOperationException());
-        try {
-            File f = File.createTempFile("deprecated", ".log", getRootDir());
-            f.deleteOnExit();
-            try (OutputStream os = new FileOutputStream(f)) {
-                getLogText().writeRawLogTo(0, os);
+        File f = super.getLogFile();
+        if (LogStorage.of(asFlowExecutionOwner()) instanceof FileLogStorage) {
+            // FileLogStorage does write a file with the same name and the same format as before JEP-210, so accept it if it is there.
+        } else {
+            // Some other storage. Cache under the traditional name, but load from the defined source on each call.
+            try {
+                f.deleteOnExit();
+                try (OutputStream os = new FileOutputStream(f)) {
+                    getLogText().writeRawLogTo(0, os);
```

Comment:
Is this call also broken for long log files for in-progress builds? It seems like it would be most commonly used for complete builds, so probably doesn't matter, but you could use the following instead:

```suggestion
                    writeLogTo(getLogText()::writeRawLogTo, os);
```